src/main/java/com/daitangroup/api/services/ProductionOrderService.java

```diff
@@ -0,0 +1,129 @@
+package com.daitangroup.api.services;
+
+import com.daitangroup.api.model.Person;
+import com.daitangroup.api.model.ProductionOrder;
+import com.daitangroup.api.model.Vehicle;
+import com.daitangroup.api.repository.PersonRepository;
+import com.daitangroup.api.repository.ProductionOrderRepository;
+import com.daitangroup.api.repository.VehicleRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class ProductionOrderService {
+
+    @Autowired
+    private PersonRepository personRepository;
+
+    @Autowired
+    private ProductionOrderRepository productionOrderRepository;
+
+    @Autowired
+    private VehicleRepository vehicleRepository;
+
+    public Optional<ProductionOrder> findProductionOrderById(long id) {
+        return productionOrderRepository.findById(id);
+    }
+
+    public Iterable<ProductionOrder> findAllProductionOrders() {
+        return productionOrderRepository.findAll();
+    }
+
+    public ProductionOrder findProductionOrderByUserId(long id) {
+
+        Optional<Person> personResult = personRepository.findById(id);
+
+        if(personResult.isPresent()) {
+            Optional<ProductionOrder> productionOrderResult = productionOrderRepository.findByPerson(personResult.get());
+            if(productionOrderResult.isPresent()) {
+                return productionOrderResult.get();
+            }
+        }
+
+        return null;
+    }
+
+    public Optional<ProductionOrder> findProductionOrderByVehicleId(long id) {
+
+        Optional<Vehicle> vehicleResult = vehicleRepository.findById(id);
+
+        if(vehicleResult.isPresent()){
+            return productionOrderRepository.findByVehicle(vehicleResult.get());
+        }
+
+        return Optional.empty();
+    }
+
+    public ProductionOrder createProductionOrder(long personid, long vehicleid, ProductionOrder productionOrder) {
```

Comment:
Can't we use the object productionOrder to pass the parameters personId and vehicleId?

Reply:
the object productionOrder only as person and vehicle, so if i want to use it to pass i will need to instantiate a vehicle and a person before. I think that this will be less efficient.